platforms/software/dependency-management/src/test/groovy/org/gradle/api/internal/attributes/DefaultAttributesSchemaTest.groovy

```diff
@@ -91,121 +92,6 @@ class DefaultAttributesSchemaTest extends Specification {
         e.message == 'Unable to find matching strategy for picard'
     }
 
-    static class CustomCompatibilityRule implements AttributeCompatibilityRule<Flavor> {
-        @Override
-        void execute(CompatibilityCheckDetails<Flavor> details) {
-            def producerValue = details.producerValue
-            def consumerValue = details.consumerValue
-            if (producerValue.name.length() > consumerValue.name.length()) {
-                // arbitrary, just for testing purposes
-                details.compatible()
-            }
-        }
-    }
-
-    static class CustomSelectionRule implements AttributeDisambiguationRule<Flavor> {
-        @Override
-        void execute(MultipleCandidatesDetails<Flavor> details) {
-            details.closestMatch(details.candidateValues.first())
-        }
-    }
-
-    static class FailingCompatibilityRule implements AttributeCompatibilityRule<Flavor> {
-        @Override
-        void execute(CompatibilityCheckDetails<Flavor> details) {
-            throw new RuntimeException()
-        }
-    }
-
-    static class FailingSelectionRule implements AttributeDisambiguationRule<Flavor> {
-        @Override
-        void execute(MultipleCandidatesDetails<Flavor> details) {
-            throw new RuntimeException()
-        }
-    }
-
-    def "merging creates schema with additional attributes defined by producer"() {
```

Comment:
These tests were moved to `ImmutableAttributesSchemaFactoryTest` as they deal with merging schemas